File: AICodingWeb/aicoding_web/app/models/stock_item_transaction.py
from datetime import datetime
from app import db
from app.models.base import BaseModel

class StockItemTransaction(db.Model, BaseModel):
    """StockItemTransaction model"""
    
    __tablename__ = 'stock_item_transactions'
    
    StockItemTransactionID = db.Column(db.Integer, primary_key=True)
    StockItemID = db.Column(db.Integer, db.ForeignKey('stock_items.StockItemID'), nullable=False)
    TransactionTypeID = db.Column(db.Integer, db.ForeignKey('transaction_types.TransactionTypeID'), nullable=False)
    CustomerID = db.Column(db.Integer, db.ForeignKey('customers.CustomerID'))
    InvoiceID = db.Column(db.Integer, db.ForeignKey('invoices.InvoiceID'))
    SupplierID = db.Column(db.Integer, db.ForeignKey('suppliers.SupplierID'))
    PurchaseOrderID = db.Column(db.Integer, db.ForeignKey('purchase_orders.PurchaseOrderID'))
    TransactionOccurredWhen = db.Column(db.DateTime, nullable=False)
    Quantity = db.Column(db.Numeric(18, 3), nullable=False)
    LastEditedBy = db.Column(db.Integer, db.ForeignKey('people.PersonID'), nullable=False)
    LastEditedWhen = db.Column(db.DateTime, nullable=False)
    
    def __repr__(self):
        return f'<StockItemTransaction {self.StockItemTransactionID}>' 

File: AICodingWeb/Models/StockItemTransaction.cs
using System;
using System.Collections.Generic;

namespace AICodingWeb.Models
{
    public class StockItemTransaction
    {
        public int StockItemTransactionID { get; set; }
        public int StockItemID { get; set; }
        public int TransactionTypeID { get; set; }
        public int? CustomerID { get; set; }
        public int? InvoiceID { get; set; }
        public int? PurchaseOrderID { get; set; }
        public int? SupplierID { get; set; }
        public DateTime TransactionOccurredWhen { get; set; }
        public decimal Quantity { get; set; }
        public int LastEditedBy { get; set; }
    }
} 